vizro-core/tests/unit/vizro/models/_action/test_action.py

```diff
@@ -282,67 +271,104 @@ def test_action_callback_function_return_value_valid(
         result = action._action_callback_function(inputs={}, outputs=callback_outputs)
         assert result == expected_function_return_value
```

Comment:
@petar-qb Are there any cases here where we don't have `expected_function_return_value == custom_action_function_mock_return()`? I don't think so, because all `_action_callback_function` does now is validate stuff.

If I'm right then let's remove the  `expected_function_return_value` and `assert` altogether and just do `action._action_callback_function(inputs={}, outputs=callback_outputs)`. If that incorrectly raises an exception then the test will fail.

I think this `test_action_callback_function_return_value_valid` has become much less valuable but is still worth keeping. Not sure we want to try all these different cases any more because some of them are quite self-evident now though.

Assuming I'm right then

Reply:
It really makes sense. I decreased the number of test cases and just leaving a few valid cases for "no outputs", "single output", "multiple list outputs" and "multiple dict outputs".